src/java.base/share/classes/java/security/spec/ECPoint.java

```diff
@@ -98,8 +99,8 @@ public boolean equals(Object obj) {
         if (this == POINT_INFINITY) return false;
 
         return obj instanceof ECPoint other
-                && ((x.equals(other.x))
-                && (y.equals(other.y)));
+                && ((Objects.equals(x, other.x))
+                && (Objects.equals(y, other.y)));
```

Comment:
`x` and `y` will not be null unless this is `POINT_INFINITY`, but this is already checked in both `equals` and `hashCode`.